src/libstd/path.rs

```diff
@@ -3801,7 +3801,7 @@ mod tests {
             });
         );
 
-        if cfg!(unix) {
+        if cfg!(unix) || cfg!(all(target_env = "sgx", target_vendor = "fortanix")) {
```

Comment:
It seems to me all other places check target_env, but only in this file target_vendor too is checked.

Reply:
There are other places where `target_vendor` is used already, like https://github.com/rust-lang/rust/blob/master/src/libstd/sys_common/mod.rs#L54

I choose `all(target_vendor = "fortanix", target_env = "sgx")` or `target_env = "sgx"` on a best-effort basis depending on the particular functionality. However, since there are no upstream targets that are `all(not(target_vendor = "fortanix"), target_env = "sgx")`, I can't really say with certainty that the conditions are correct everywhere.